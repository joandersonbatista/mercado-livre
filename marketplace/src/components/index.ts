export { Header } from './header/header';
export { IconButton } from './icon-button/icon-button';
export { Button } from './button/button';
export { Breadcrumbs } from './breadcrumbs/breadcrumbs';
export { BreadcrumbsSkeleton } from './breadcrumbs/breadcrumbs-skeleton';
export { Skeleton, type SkeletonProps } from './skeleton/skeleton';
export { LanguageSwitcher } from './language-switcher/language-switcher';
export { ServerError } from './server-error/server-error';
