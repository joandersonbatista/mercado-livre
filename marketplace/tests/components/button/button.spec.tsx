import { Button } from '@/components';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Button Component', () => {
  it('should render the button with the provided children', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });

  it('should apply the className correctly', () => {
    render(<Button className="custom-class">Click me</Button>);

    const buttonElement = screen.getByText('Click me');

    expect(buttonElement).toHaveClass('button');
    expect(buttonElement).toHaveClass('custom-class');
  });

  it('should forward additional props to the button element', () => {
    const handleClick = jest.fn();

    render(
      <Button disabled onClick={handleClick}>
        Click me
      </Button>,
    );

    fireEvent.click(screen.getByText('Click me'));
    expect(handleClick).not.toHaveBeenCalled();
    expect(screen.getByText('Click me')).toBeDisabled();
  });

  it('should trigger onClick when clicked', () => {
    const handleClick = jest.fn();

    render(<Button onClick={handleClick}>Click me</Button>);

    fireEvent.click(screen.getByText('Click me'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should match the snapshot', () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container).toMatchSnapshot();
  });
});
