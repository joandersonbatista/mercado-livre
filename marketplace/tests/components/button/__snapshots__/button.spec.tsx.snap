// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match the snapshot 1`] = `
<div>
  <button
    class="button "
  >
    Click me
  </button>
</div>
`;
